runtime/bcutil/defineclass.c

```diff
@@ -764,6 +764,7 @@ callDynamicLoader(J9VMThread *vmThread, J9LoadROMClassData *loadData, U_8 * inte
 	/* The module of a class transformed by a JVMTI agent needs access to unnamed modules */
```

Comment:
Can you modify this to check `result` once around the outside of this if and the next?

Something like:
```c
if (BCT_ERR_NO_ERROR == result) {
    if ((J2SE_VERSION(vm) >= J2SE_V11)...... {
    .....
    if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_RECREATE_CLASSFILE_ONLOAD)) {
    }   
}
```
as the `J9_EXTENDED_RUNTIME_RECREATE_CLASSFILE_ONLOAD` immediately checks for `BCT_ERR_NO_ERROR` as well

Reply:
Updated. I expect the `(NULL != loadData->romClass)` is redundant and can be removed as well, what do you think?

I was trying to get a JVM created for the user to try. There may be more than one problem, we'll see what happens on a re-run with this change. The machines were busy with nightly testing and I missed the user's timezone window last night.

Reply:
I think you're right.  I wonder if we added the null check to satisfy one of the static analysis tools we've run

Reply:
I figured we needed the null check in the absence of checking the result code.